modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/RaftGroupServiceImpl.java

```diff
@@ -373,7 +374,26 @@ public static CompletableFuture<RaftGroupService> start(
 
         sendWithRetry(leader, req, currentTimeMillis() + timeout, fut);
 
-        return fut.thenRun(() -> {});
+        return fut.handleAsync((resp, err) -> {
```

Comment:
It's required to guard such async logic with corresponding busy lock, otherwise we might finish with stopped executor.

Reply:
Correct me please if I'm wrong, you suggest to add busy lock as a parameter to `RaftGroupService#changePeersAsync`, so we can check busy lock of a caller in the async logic, right?